plugins/templates/create.js

```diff
@@ -21,38 +23,42 @@ module.exports = () => ({
                 security: [{ token: [] }]
             }
         },
-        handler: (request, reply) => {
-            const pipelineFactory = request.server.app.pipelineFactory;
-            const templateFactory = request.server.app.templateFactory;
-            const pipelineId = request.auth.credentials.pipelineId;
-            const name = request.payload.name;
-            const labels = request.payload.labels || [];
+        handler: (request, reply) => validator(request.payload.yaml)
+            .then((config) => {
+                if (config.errors.length > 0) {
+                    throw boom.badRequest(
+                        'Template has invalid format: ', config.errors.toString());
+                }
 
-            return Promise.all([
-                pipelineFactory.get(pipelineId),
-                templateFactory.list({ params: { name } })
-            ]).then(([pipeline, templates]) => {
-                const templateConfig = hoek.applyToDefaults(request.payload, {
-                    pipelineId: pipeline.id,
-                    labels
-                });
+                const pipelineFactory = request.server.app.pipelineFactory;
+                const templateFactory = request.server.app.templateFactory;
+                const pipelineId = request.auth.credentials.pipelineId;
 
-                // If template name doesn't exist yet, just create a new entry
-                if (templates.length === 0) {
-                    return templateFactory.create(templateConfig);
-                }
+                return Promise.all([
+                    pipelineFactory.get(pipelineId),
+                    templateFactory.list({ params: { name: config.template.name } })
+                ]).then(([pipeline, templates]) => {
+                    const templateConfig = hoek.applyToDefaults(config.template, {
+                        pipelineId: pipeline.id,
+                        labels: config.template.labels || []
```

Comment:
Are we really making labels part of the template definition? That doesn't seem right. This used to be part of the original payload.